Fix use case for Star Gem in console mode

michalpelka opened the pull request:
## What does this PR do?

It fixes issue described here : https://github.com/o3de/o3de/issues/18474

## How was this PR tested?

- Cherry picked against 24.09
- ROS 2 Gem enabled
- ROS 2 Camera added
- Visualize camera in RQT
![image](https://github.com/user-attachments/assets/77be2c31-7b14-4bcf-bd2b-87d001636f6c)

moudgils commented on Gems/Stars/Code/Source/StarsFeatureProcessor.cpp:
```diff
@@ -60,11 +60,16 @@ namespace AZ::Render
+            m_viewportSize = viewportContext->GetViewportSize();
+            RPI::ViewportContextIdNotificationBus::Handler::BusConnect(viewportContext->GetId());
+        }
+        else{
```
```suggestion
        else
        {
```

moudgils commented on Gems/Stars/Code/Source/StarsFeatureProcessor.cpp:
```diff
@@ -60,11 +60,16 @@ namespace AZ::Render
+            RPI::ViewportContextIdNotificationBus::Handler::BusConnect(viewportContext->GetId());
+        }
+        else{
+            m_viewportSize = AzFramework::WindowSize(1280, 720);
```
I wonder if we want to use r_width/r_height cars instead of hardcoded 720p res?

michalpelka commented:
@moudgils as @spham-amzn suggested I removed magic number (1280 by 720). I've moved minimal size of window to cpp namespace and reused in proposed code.

michalpelka commented:
Rebased, I will re-triger jenkins.

nick-l-o3de commented:
This suffers the current problem on Android
```
D:/workspace/o3de/Gems/Stars/Code/Source/StarsFeatureProcessor.cpp(73,72): error C2220: the following warning is treated as an error [D:\workspace\o3de\build\windows\External\Stars-97f44601\Code\Stars.Static.vcxproj]
[2024-11-19T19:33:15.653Z] D:/workspace/o3de/Gems/Stars/Code/Source/StarsFeatureProcessor.cpp(73,72): error C2220:             m_viewportSize = AzFramework::WindowSize(MinViewPortWidth, MinViewPortHeight); [D:\workspace\o3de\build\windows\External\Stars-97f44601\Code\Stars.Static.vcxproj]
[2024-11-19T19:33:15.653Z] D:/workspace/o3de/Gems/Stars/Code/Source/StarsFeatureProcessor.cpp(73,72): error C2220:                                                                        ^ [D:\workspace\o3de\build\windows\External\Stars-97f44601\Code\Stars.Static.vcxproj]
[2024-11-19T19:33:16.216Z] D:/workspace/o3de/Gems/Stars/Code/Source/StarsFeatureProcessor.cpp(73,72): warning C4244: 'argument': conversion from 'const float' to 'const uint32_t', possible loss of data [D:\workspace\o3de\build\windows\External\Stars-97f44601\Code\Stars.Static.vcxproj]
```

michalpelka commented:
fixed using static cast and build on Windows. Triggering AR again.
